### ISSUE TITLE
Fix S2190 FN: Support mathematical operators in generic types

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6385]
### Description

FN for S2190 (InfiniteRecursion).

Using C#11 static mathematical operator overload, the rule is not raising an issue.

### Repro steps

```csharp
      public class Addition : IAdditionOperators<Addition, Addition, Addition>
      {
          public static Addition operator +(Addition left, Addition right) => left + right; // FN
      }
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
